server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -437,10 +427,14 @@ public DocumentField field(String fieldName) {
     * */
     public void setDocumentField(String fieldName, DocumentField field) {
         if (fieldName == null || field == null) return;
-        if (documentFields.size() == 0) this.documentFields = new HashMap<>();
         this.documentFields.put(fieldName, field);
     }
 
+    public void setDocumentFields(Map<String, DocumentField> docFields, Map<String, DocumentField> metaFields) {
```

Comment:
`addDocumentFields`? It "adds" rather than "sets". And some day it's name is going to confuse me.

Reply:
++